### ISSUE TITLE
ARM is taking 7.5 hours to rip single DVD

### ISSUE THREAD

**Issue by mpompey** [automatic-ripping-machine/automatic-ripping-machine#313]
Hey guys, let me first start off by saying thank you for putting this project together. I run a DIY home lab and am learning alot about VMs, ESXI, etc through this and many other projects. So once again, thank you.

Now my issue. My machine takes anywhere from 7 - 9 hours to rip a single DVD. I'm currently running ARM on an ubuntu 16.04 VM with 8GB RAM, 4 vCPUs, 3 DVD-ROMs (pass through from host) and 100GB of HDD space on an ESXi 6.0 host.  My DIY ESXI host box is a white build with an Intel i5 CPU, 16GB RAM, and 1TB HDD. This is the only VM running on it. I've tried searching for this issue online and haven't found anything exactly what I' experiencing.
 
From what I've read online, it should take maybe 2 hours to rip a simple DVD with an i5 CPU. I'm not sure if it is related but often ARM will rip 2-3 copies of the DVD into separate folders under /home/arm/media/notidentified.

I also have an ubuntu 18.04 VM with the same hardware settings and it takes the same time to rip. In addition, while monitoring the VM while it is ripping, it will barely use any of the 8GB of RAM allocated to it, but the CPU is clipped at 100% during the entire rip. Not sure if that is to be expected or if something is off. Never mind trying to get it to rip of sr1 or sr2

Machine Specs:
DIY ESXi box
Intel i5 CPU
16GB RAM
Gigabyte H170M DS3H MOBO
3 48X CD/DVD-Rom drives
HDD Western Digital 1 TB BLue

VM ubuntu 16.04
4 vCPUs
8GB RAM
100GB Virtual HD

I've attached the DEBUG version of my log while ripping Aliens V Predator DVD (No Judgement!). I've also attached a copy of my /opt/arm/arm.yaml config file to see if maybe I have something misconfigured.

### Environment
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.6 LTS"

ARM Release Version 2.1 dev

### Log file
[Run the rip in DEBUG and drag and drop the log file onto this comment box]

[>>>>> DRAG AND DROP LOG FILE HERE <<<<<]
[ALIEN_VS_PREDATOR.log](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/3757870/ALIEN_VS_PREDATOR.log)

[arm_yaml.txt](https://github.com/automatic-ripping-machine/automatic-ripping-machine/files/3757898/arm_yaml.txt)

**Comment by mpompey** [comment]
Update: Some progress, tweaked a couple of things

I turned Main Feature to "True". This brought down the DVD rip time to about an hour and 15 minutes. The sample DVD in question was Austin Powers #1. It resulted in 1 mkv file being being created vs the three separate folders of the exact same rips.

Second thing I did was install additional ACHI to the ESXi box from the vibsdepot. I was wondering if ESXi had removed some of the mobo ACHI drivers and it was defaulting to a generic driver. But I'm not sure if that had any effect as I was unable to passthrough the ACHI controller to the VM underneath.

As I have way more DVDs than blu-rays in my collection, I'll focus on them first as I continue to tweak and play with my setup. Next task, figuring why I can only successfully rip from DVD-ROM at sr0 versus the other 2.

**Comment by mpompey** [comment]
Another update:

Fixed issue: only ripping from sr0
Turns out I inadvertently passed through the same DVD-ROM drive 3X to the VM. I had assumed that ESXi would automatically route the appropriate hardware device to the underlying VM. But turns out there was another step I needed to take after adding a CD-ROM, specifying exactly which physical rom device was to be connected. Once I figured that out I was able to start ripping from sr1 and sr2

Fixed issue: CPU utilization
This fix brought down ripping time, as the VM wasn't trying to rip the same 3 times simultaneously. At the same time CPU utilization dropped to something more appropriate as I wasn't redlining the system trying to rip the same dvd through 3 different mount points.  

Fixed issue: Handbrake riping multiple copies
The other issue related to disc ripping time is related to Disney DVDs that I own. I've been learning about DRM methods and one is the dreaded 99-Titles. Handbrake was ripping every single one of the false titles and that was taking a long time. So I need to continue research for that, or skip over the Disney DVDs in my collection. I would get anywhere from 6 to 40 different mkv rips in the completed folder but each one would have chapters scrambled, audio out of sync, except for one. And trying to figure out which one it is has been a pain so far. Need to continue my homework for this issue.
